spec/memo_wise_spec.rb

```diff
@@ -357,5 +357,208 @@ def self.class_method; end
+              stub_const("ModuleWithMemo", module_with_memo)
+            end
+
+            let(:class_extending_module_with_memo) do
+              Class.new do
+                extend ModuleWithMemo
+              end
+            end
+
+            let(:target) { class_extending_module_with_memo }
+
+            it_behaves_like "#memo_wise shared examples"
+          end
+
+          context "when 1 module extended by 2 classes" do
+            let(:module_with_memo) do
+              Module.new do
+                prepend MemoWise
+
+                def test_method
+                  Random.rand
+                end
+                memo_wise :test_method
+              end
+            end
+            let(:class_a_extending_module_with_memo) do
+              Class.new do
+                extend ModuleWithMemo
+              end
+            end
+            let(:class_b_extending_module_with_memo) do
+              Class.new do
+                extend ModuleWithMemo
+              end
+            end
+
+            before(:each) do
+              stub_const("ModuleWithMemo", module_with_memo)
+            end
+
+            it <<~DESC do
+            does memoize class method for both classes
+            with same name with separate caches
+            DESC
+              aggregate_failures do
+                expect(class_a_extending_module_with_memo.test_method).
+                  to eq(class_a_extending_module_with_memo.test_method)
+                expect(class_b_extending_module_with_memo.test_method).
+                  to eq(class_b_extending_module_with_memo.test_method)
+                expect(class_a_extending_module_with_memo.test_method).
+                  to_not eq(class_b_extending_module_with_memo.test_method)
+              end
+            end
+          end
+        end
+
+        context "when included" do
+          context "when defined with 'def'" do
+            include_context "with context for module methods via normal scope"
+
+            before(:each) do
+              stub_const("ModuleWithMemo", module_with_memo)
+            end
+
+            let(:class_extending_module_with_memo) do
+              Class.new do
+                include ModuleWithMemo
+              end
+            end
+            let(:instance) { class_extending_module_with_memo.new }
+
+            let(:target) { instance }
+
+            it_behaves_like "#memo_wise shared examples"
+          end
+
+          context "when defined with 'def self.' and 'def'" do
+            let(:module_with_memo) do
+              Module.new do
+                prepend MemoWise
+
+                def self.test_method
+                  Random.rand
+                end
+                memo_wise self: :test_method
+
+                def test_method
+                  Random.rand
+                end
+                memo_wise :test_method
+              end
+            end
+            let(:class_extending_module_with_memo) do
```

Comment:
```suggestion
            let(:class_including_module_with_memo) do
```

This one `include`s rather than `extend`s, right?